File: Gnome_sort/Ruby/jcla1/gnome_sort_test.rb

require_relative "./gnome_sort"

describe "#gnome_sort" do
  it "Sorts empty array" do
    arr = []
    gnome_sort(arr).should eq(arr)
  end

  it "Returns unchanged array with 1 element" do
    arr = [1]
    gnome_sort(arr).should eq(arr)
  end

  it "Sorts array" do
    arr = [9, 4, 6, -2, 6, -10, 592, 41]
    gnome_sort(arr).should eq(arr.sort)
  end
end

File: Gnome_sort/Java/ghostsnstuff/GnomeSortTest.java
// output
// 5 3 2 4 8 7 1  # pre-sort
// 1 2 3 4 5 7 8  # post-sort 
class GnomeSortTest extends GnomeSort {
  public static void main(String[] args) {
    int[] arr = {5,3,2,4,8,7,1};
    log(arr);
    System.out.println(" # pre-sort");
    arr = GnomeSort(arr);
    log(arr);
    System.out.println(" # post-sort ");	
  }
}
